enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,372 @@
+```yaml
+apiVersion: config.openshift.io/v1 
+kind: ImageDigestMirrorSet 
+metadata:
+  name: ubi9repo
+spec:
+  imageDigestMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    - example.com/example2/ubi-minimal
+    source: registry.access.redhat.com/ubi9/ubi-minimal 
+    mirrorSourcePolicy: AllowContactingSource 
+  - mirrors:
+    - mirror.example.com
+    source: registry.redhat.io
+    mirrorSourcePolicy: NeverContactSource
+```
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageTagMirrorSet
+metadata:
+  name: tag-mirror
+spec:
+  imageTagMirrors:
+  - mirrors:
+    - docker.io
+    source: docker-mirror.internal
+    mirrorSourcePolicy: AllowContactingSource
+```
+
+This would result in WMCO generating content for 3 files and transferring them onto each Windows instance.
+The file structure would be as follows:
+```powershell
+$ tree $config_path
+C:/k/containerd/registries/
+|── registry.access.redhat.com
+|   └── hosts.toml
+|── mirror.example.com
+|   └── hosts.toml
+└── docker.io
+    └── hosts.toml
+```
+
+And the content for each generated file would look like this: 
+```powershell
+$ cat "$config_path"/registry.access.redhat.com/host.toml
+server = "https://registry.access.redhat.com" # default fallback server since "AllowContactingSource" mirrorSourcePolicy is set
+
+[host."https://example.io/example/ubi-minimal"]
+ capabilities = ["pull"]
+
+[host."https://example.com/example2/ubi-minimal"] # secondary mirror
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/registry.redhat.io/host.toml
+# "server" omitted since "NeverContactSource" mirrorSourcePolicy is set
+
+[host."https://mirror.example.com"]
+ capabilities = ["pull"]
+
+
+$ cat "$config_path"/docker.io/host.toml
+server = "https://docker.io"
+
+[host."https://docker-mirror.internal"]
+ capabilities = ["pull", "resolve"] # resolve tags
+```
+
+---
+
+## Test Plan
+
+In addition to unit testing individual WMCO/WICD packages and controllers, an e2e job will be added to the release repo
+for WMCO's master/release-4.15 branches. A new CI workflow will be created leveraging 
+[existing step-registry steps](https://github.com/openshift/release/tree/master/ci-operator/step-registry/aws/provision/vpc/disconnected),
+which creates a disconnected cluster in AWS with hybrid-overlay networking. 
+I propose AWS here instead of vSphere since we have 3+ existing jobs that run on vSphere and I don't want to overload the infra.
+We'll also need to pre-load the images used in our e2e suite into the cluster's internal registry as part of job setup.
+This workflow will be used to run the existing WMCO e2e test suite to ensure that WMCO runs as expected in disconnected 
+environments with no regression. We will add a few test cases to explicitly check the state of registry settings on Windows
+nodes.
+QE should cover disconnected environments on all supported platforms when validating this feature.
+
+### User Stories
+
+Story 1: Set `config_path` in containerd_conf.toml
```

Comment:
The user stories should be of the form: `As an OpenShift Windows Admin, I should be able to define mirrors for my cluster, so my cluster would function when  disconnected from primary registries`.

Reply:
ACK, I've added in details in the proper form.